src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_cache_impl.cc

```diff
@@ -60,11 +68,54 @@ smart_objects::SmartObject MergeModuleData(
   for (; it != data2.map_end(); ++it) {
     const std::string& key = it->first;
     const smart_objects::SmartObject& value = it->second;
-    result[key] = value;
+    // Merge maps and arrays with `id` param included, replace other types
+    if (result.keyExists(key) && value.getType() == result[key].getType()) {
+      if (value.getType() == smart_objects::SmartType::SmartType_Map) {
+        result[key] = MergeModuleData(result[key], value);
+      } else if (value.getType() == smart_objects::SmartType::SmartType_Array) {
+        result[key] = MergeArray(result[key], value);
+      } else {
+        result[key] = value;
+      }
+    } else {
+      result[key] = value;
+    }
   }
   return result;
 }
 
+smart_objects::SmartObject MergeArray(const smart_objects::SmartObject& data1,
+                                      const smart_objects::SmartObject& data2) {
+  // Merge data only in the case where each value in the array is an Object with
+  // an ID included, otherwise replace
+  if (!data2.empty() &&
+      data2.getElement(0).getType() ==
+          smart_objects::SmartType::SmartType_Map &&
+      data2.getElement(0).keyExists(application_manager::strings::id)) {
+    smart_objects::SmartObject result = data1;
+    smart_objects::SmartArray* result_array = result.asArray();
+    smart_objects::SmartArray* data_array = data2.asArray();
+    auto data_it = data_array->begin();
+    for (; data_it != data_array->end(); ++data_it) {
+      auto result_it =
+          std::find_if(result_array->begin(),
+                       result_array->end(),
+                       [data_it](smart_objects::SmartObject& obj) -> bool {
+                         return obj[application_manager::strings::id] ==
+                                (*data_it)[application_manager::strings::id];
+                       });
+
```

Comment:
@jacobkeeler 
```cpp
if (result_contains((*data_it)[application_manager::strings::id])) {
   *result_it = MergeModuleData(*result_it, *data_it);
} else {
  result_array->push_back(*data_it);
}
```

Reply:
I used a different form of lambda, so this no longer applies